### DISCUSSION TITLE
Use MagicLink to login user for e2e test

### DISCUSSION THREAD

**Discussion by cMoney667** [pbteja1998/remix-auth-email-link#12] [Q&A]
I have been doing some testing to see if there is a way I could authenticate the user via the email link.

I have accessed the session variable and tried to redirect the user via a test route but when it gets their it says it is malformed UTF-8. Was wondering if there was a different way someone else might have been able to get this to work properly for their tests

Here is a link to the example repository, https://github.com/shortnd/remix-email-auth-test

the test page is at `__tests/login`

**Comment by TrangPham** [comment]
Hi I am also wondering how I can login a user or set a user as logged in for E2E testing. I am using cypress.
